lib/commands/battery.js

```diff
@@ -8,10 +10,26 @@ export default {
    * @this {import('../driver').XCUITestDriver}
    */
   async mobileGetBatteryInfo() {
-    return /** @type {BatteryInfo} */ (await this.proxyCommand('/wda/batteryInfo', 'GET'));
+    let batteryInfoFromShimService;
+    if (isIos18OrNewer(this.opts)) {
+      // TODO: Remove this once we publish the appium-ios-remotexpc package to npm
+      /* eslint-disable import/no-unresolved */
+      // @ts-expect-error
+      const {Services} = await import('appium-ios-remotexpc');
+      const diagService = await Services.startDiagnosticsService(this.device.udid);
+      batteryInfoFromShimService = await diagService.ioregistry({
+        ioClass: 'IOPMPowerSource',
+        returnRawJson: true,
+      });
+    }
+
+    const batteryInfoFromWda = /** @type {import('./battery-types.js').BatteryInfo} */ (
+      await this.proxyCommand('/wda/batteryInfo', 'GET')
+    );
+    return {...batteryInfoFromWda, ...batteryInfoFromShimService};
```

Comment:
please also update the documentation at https://github.com/appium/appium-xcuitest-driver/blob/master/docs/reference/execute-methods.md#mobile-batteryinfo